### ISSUE TITLE
include more complex tests

### ISSUE THREAD

**Issue by jakobj** [INM-6/h5py_wrapper#7]
as far as i can see, our test dictionary is quite simple. it might be a good idea to include more complex examples (e.g., lists of dicts, sets, tuples)

**Comment by mschmidt87** [comment]
My opinion on this is that instead of trying to take of all kinds of weird kinds of datasets, we should rather stick to the simple test/examples and wait for users to approach with use cases. 
The examples you mentioned might still be sensible and will partly be solved and then also tested in the PR fixing #11 .

**Comment by jakobj** [comment]
agreed. no need to make up and handcraft use cases. we can consider using [hypothesis](http://hypothesis.works/)
